File: Python/Aksesoris.py
from Petshop import Petshop

class Aksesoris(Petshop):
    def __init__(self, id=0, nama_produk="", harga_produk=0, stok_produk=0, jenis="", bahan="", warna=""):
        # Inisialisasi atribut-atribut dengan nilai awal beserta init dari parent class
        super().__init__(id, nama_produk, harga_produk, stok_produk)
        self.__jenis = jenis
        self.__bahan = bahan
        self.__warna = warna
    
    # Setter methods buat setiap atribut
    def set_jenis(self, jenis):
        self.__jenis = jenis
    
    def set_bahan(self, bahan):
        self.__bahan = bahan
    
    def set_warna(self, warna):
        self.__warna = warna
    
    # Getter methods buat setiap atribut
    def get_jenis(self):
        return self.__jenis
    
    def get_bahan(self):
        return self.__bahan
    
    def get_warna(self):
        return self.__warna
    
    # Method untuk menampilkan data beserta unsur tabel dinamis, override (super) dari fungsi di parent class
    def tampilkan_data(self, panjang_id, panjang_nama, panjang_harga, panjang_stok, panjang_jenis, panjang_bahan, panjang_warna):
        super().tampilkan_data(panjang_id, panjang_nama, panjang_harga, panjang_stok)
        print(f" | {self.__jenis}", end="")
        for i in range(panjang_jenis - len(self.__jenis)):
            print(" ", end="")
        print(f" | {self.__bahan}", end="")
        for i in range(panjang_bahan - len(self.__bahan)):
            print(" ", end="")
        print(f" | {self.__warna}", end="")
        for i in range(panjang_warna - len(self.__warna)):
            print(" ", end="")

File: Java/Aksesoris.java
public class Aksesoris extends Petshop {
    private String jenis;
    private String bahan;
    private String warna;

    public Aksesoris() {
        super();
        this.jenis = "";
        this.bahan = "";
        this.warna = "";
    }

    public Aksesoris(int id, String namaProduk, int hargaProduk, int stokProduk, String jenis, String bahan, String warna) {
        super(id, namaProduk, hargaProduk, stokProduk);
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
    }

    // Setter methods
    public void setJenis(String jenis) { this.jenis = jenis; }
    public void setBahan(String bahan) { this.bahan = bahan; }
    public void setWarna(String warna) { this.warna = warna; }

    // Getter methods
    public String getJenis() { return jenis; }
    public String getBahan() { return bahan; }
    public String getWarna() { return warna; }

    public void tampilkanData(int panjangId, int panjangNama, int panjangHarga, int panjangStok, int panjangJenis, int panjangBahan, int panjangWarna) {
        super.tampilkanData(panjangId, panjangNama, panjangHarga, panjangStok);
        System.out.print(" | " + jenis);
        for (int i = 0; i < panjangJenis - jenis.length(); i++) System.out.print(" ");
        System.out.print(" | " + bahan);
        for (int i = 0; i < panjangBahan - bahan.length(); i++) System.out.print(" ");
        System.out.print(" | " + warna);
        for (int i = 0; i < panjangWarna - warna.length(); i++) System.out.print(" ");
    }
}